### PR TITLE
Translated code examples from 01-intro to pytorch

### PR DESCRIPTION
**Pull request by danishpruthi**

### CODE REVIEW

**Comment by neubig** [comment]
I'd like to merge this, but perhaps it'd be good to have one of the other TAs do a code review of the PyTorch examples first? Also it would be nice to run both the DyNet and PyTorch examples and make sure they both converge and get similar performance.

**Comment by danishpruthi** [comment]
Yes, I agree. I have requested Vaibhav to review the examples that I convert and I can review his. 

Regarding performance, I found pytorch to be considerably slower (it takes close to 6-7 sec/iteration), right now there is no batching (so that students can follow easily), and the performance is slightly lower (by 2-4%). The implementation is not very complicated so there are lesser chances of bugs, but will wait for Vaibhav to double check.

File: 01-intro-pytorch/model.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        use_cuda = torch.cuda.is_available()
+
+        if use_cuda:
+            type = torch.cuda.FloatTensor
+
+        self.bias = Variable(torch.zeros(ntags),
+                             requires_grad=True).type(type)
+
+        """ layers """
+        self.embedding = nn.Embedding(nwords, ntags)
```
Comment (MysteryVaibhav) [inline 01-intro-pytorch/model.py:20]: I think it's just a good practice to initialize the parameters in pytorch, specially the embedding matrix.

**Review by MysteryVaibhav — COMMENTED** [review]
@danishpruthi I have reviewed the code. The code seems conceptually identical to the dynet version. I just had a suggestion about adding initialization of parameters. Not a mandatory change though, if you decide to add it then good, otherwise it can be merged as is.

File: 01-intro-pytorch/model.py (hunk 1 above)
Comment (danishpruthi) [reply 01-intro-pytorch/model.py:20]: Thanks @MysteryVaibhav, I have fixed the problem. The accuracies are also better and at par with dynet. The dynet defualt initialization is Glorot initialization (Glorot, X. & Bengio, Y. (2010)). So I used the same.

**Comment by danishpruthi** [comment]
@neubig: I think you can merge this now. Vaibhav reviewed this, and the numbers match.

**Pull request merged**